examples/inverse_folding/README.md

```diff
@@ -0,0 +1,170 @@
+
+## General usage
+
+### Load model
+The `esm_if1_gvp4_t16_142M_UR50` function loads the pretrained model and its
+corresponding alphabet. The alphabet represents the amino acids and the special
+tokens encoded by the model.
+
+```
+import esm
+model, alphabet = esm.pretrained.esm_if1_gvp4_t16_142M_UR50()
+```
+
+### Input format
+The input to the model is a list of backbone atom coordinates for the N, CA, C
+atoms in each amino acid. For each structure, the coordinate list `coords` would
+be of shape L x 3 x 3, where L is the number of amino acids in the structure. 
+`coords[i][0]` is the 3D coordinate for the N atom in amino acid `i`, 
+`coords[i][1]` is the 3D coordinate for the CA atom in amino acid `i`, and
+`coords[i][2]` is the 3D coordinate for the C atom in amino acid `i`. 
+
+### Load input data from PDB and mmCIF file formats
+To load a structure from PDB and mmCIF file formats and extract the backbone
+coordinates of the N, CA, C atoms as model input,
+```
+import esm
+structure = esm.inverse_folding.util.load_structure(fpath, chain_id)
+coords, seq = esm.inverse_folding.util.extract_coords_from_structure(structure)
+```
+
+### Example Jupyter notebook
+See `examples/inverse_folding/notebook.ipynb` for examples of sampling sequences, 
+calculating conditional log-likelihoods, and extracting encoder output as
+structure representation.
+
+This notebook is also available on colab:
+  
+[<img src="https://colab.research.google.com/assets/colab-badge.svg">](https://colab.research.google.com/github/facebookresearch/esm/blob/master/examples/inverse_folding/notebook.ipynb)
+
+### Sample sequence designs
+To sample sequences for a given set of backbone coordinates,
+```
+sampled_seq = model.sample(coords, temperature=T)
+```
+
+The temperature parameter controls the ``sharpness`` of the probability
+distribution for sequence sampling. Higher sampling temperatures yield more
+diverse sequences but likely with lower native sequence recovery.
+The default sampling temperature is `T=1`. To optimize for native sequence
+recovery, we recommend sampling with low temperature such as `T=1e-6`.
+
+### Scoring sequences
+To score the conditional log-likelihoods for sequences conditioned on a given
+set of backbone coordinates, use the `score_sequence` function,
+```
+import esm
+ll_fullseq, ll_withcoord = esm.inverse_folding.util.score_sequence(model, alphabet, coords, seq)
+```
+
+The first returned value ``ll_fullseq`` is the average log-likelihood averaged
+over all amino acids in a sequence.
+The second return value ``ll_withcoord`` is averaged only over those amino acids
+with associated backbone coordinates in the input, i.e., excluding those with
+missing backbone coordinates.
+
+### Partially masking backbone coordinates
+To mask a parts of the input backbone coordinates, simply set those coordinate
+values to `np.inf`. For example, to mask the backbone coordinates for the first
+ten amino acid in the structure,
+```
+coords[:10, :] = np.inf
```

Comment:
nit: `float('inf')` -- i guess coords should be a torch.tensor right. better leave numpy alone then :)

Reply:
Done, thanks!